### PR TITLE
MySQL backup app

### PR DESCRIPTION
**Pull request by earlruby**
Sample app to demonstrate how to backup a MySQL service to an NFS mount point on an hourly basis.

Includes a monitoring route that tells you how many seconds since the last backup.

@zquestz @variadico 

cc: @lparis

### CODE REVIEW

File: mysql-backup/mysql-backup.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    --password=$URI_PW --port=$URI_PORT \
+    --skip-opt --add-drop-database --add-drop-table \
+    --add-locks --allow-keywords --create-options \
+    --comments --complete-insert --compress \
+    --create-options --routines --tz-utc \
+    --disable-keys --extended-insert --single-transaction \
+    --quick --routines --tz-utc --set-charset \
+    $DBNAME | gzip > $DB_BACKUP_DIR/mysql-${DBNAME}-backup.$DATE.gz
+
+touch /tmp/backup.complete
```
Comment (zquestz) [inline mysql-backup/mysql-backup.sh]: NIT: Need trailing newline.

File: mysql-backup/bash_start.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+DB_BACKUP_DIR=/backups/mysql-service
+
+# Make sure that the backup script is executable
+sudo chmod +x /app/mysql-backup.sh
+
+echo "==> INSTALLING mysql-client"
+
+# Install the MySQL client software, including mysqldump
+sudo apt-get update
+sudo apt-get install mysql-client -y
```
Comment (zquestz) [inline mysql-backup/bash_start.sh]: This is fine for now, but I think we should put together a mysql-client package.

**Comment by zquestz** [comment]
LGTM

File: mysql-backup/mysql-backup.sh (hunk 1 above)
Comment (earlruby) [inline mysql-backup/mysql-backup.sh]: Done.

File: mysql-backup/bash_start.sh (hunk 2 above)
Comment (earlruby) [inline mysql-backup/bash_start.sh]: I really just wanted a general purpose mysql-client package I could merge into any app that needed those libs and programs. We should look at doing this for our general-purpose Docker package stager for v2.

**Comment by earlruby** [comment]
@wallyqs

File: mysql-backup/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+### mysql-backup
+
+This is a sample bash application that runs two scripts, as accepted by Apcera Platform's built-in bash stager:
+
+1. `bash_start.sh` installs the MySQL client software, creates a crontab file `/etc/cron.d/backup`, starts a monitoring loop, and starts cron.
+2. `mysql-backup.sh` is executed once an hour by cron. It backs up the MySQL server that's bound to the app and saves the backup onto the NFS service that's bound to the app.
```
Comment (variadico) [inline mysql-backup/README.md]: _super_ nit: I'd choose either `_` or `-` for a word separator. Not both.

Comment (earlruby) [inline mysql-backup/README.md]: Good catch actually. I always use `-` for script names, `_` for bash variables, but `bash_start.sh` is a "magic" name that the platform looks for. 

I'll change it.

**Comment by earlruby** [comment]
@variadico: The change you asked for has been added.

**Comment by variadico** [comment]
LGTM

**Pull request merged**